livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/utils.py

```diff
@@ -67,7 +67,7 @@ def _build_oai_image_content(image: llm.ChatImage, cache_key: Any):
     if isinstance(image.image, str):  # image url
         return {
             "type": "image_url",
-            "image_url": {"url": image.image, "detail": "auto"},
+            "image_url": {"url": image.image, "detail": image.inference_detail},
```

Comment:
This should be done automatically depending on the inference_width & inference_height. 

Tho we don't have tests covering this today so it may not work as expected

Reply:
I'm not so sure...  there are legitimate use-cases for high detail on small images, which on a 512x512 image will use 255 tokens vs 85 tokens for low detail, and gives better results.  Low detail on large images is less useful of course..., and you should probably resize it to 512x512 yourself first to improve inference speed.

this class as it is is a little messy, since inference_width/height are only used for video frames and ignored for URLs, and the actual API capabilities in the underlying models vary quite a bit (I'm not aware of any other provider with the "detail" parameter, for instance).

Reply:
From my understanding, putting low basically resize the image to 512x512. So having an input image of this size is the same as detail "low"
https://platform.openai.com/docs/guides/vision#low-or-high-fidelity-image-understanding

Reply:
> high will enable "high res" mode, which first allows the model to first see the low res image (using 85 tokens) and then creates detailed crops using 170 tokens for each 512px x 512px tile.

Ah ok I guess the behaviour is different, then it lgtm

Reply:
yeah i assumed the same based on the docs but it isn't what i observe in testing. while it is true that passing "low" will resize your image to 512x512, the docs don't explicitly say the reverse is true. it only says 

> By default, the model will use the auto setting which will look at the image input size and decide if it should use the low or high setting.

That doesn't specify where the threshold to flip to "low" is, and my testing suggests this threshold is much lower than 512x512. in "high" mode each 512x512 tile gets a 170tk budget on top of the whole image's 85tk budget for the 512x512 thumbnail, which does imply that using "high" on a 512x512 image has value and provides deeper understanding than using "low".